### PR TITLE
Added logic to prematurely return on custom actions

### PR DESCRIPTION
**Pull request by LuigiPulcini**
This change adds the possibility for custom actions to return prematurely before the AJAX request is submitted.

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
The `do_variation_action` method in `wc_meta_boxes_product_variations_ajax` doesn't offer any possibility for a custom action to interrupt the script before the AJAX request is sent. For this reason, any custom action that uses a `window.confirm` or a `window.prompt`, doesn't have a way to stop the operation if the user clicks Cancel on those dialogs.

The proposed change offers the developer the possibility to set the `cancel` property of the `data` object passed to the `do_variation_action + '_ajax_data'` handler callback so that, when `cancel === true`, the method returns prematurely without submitting the AJAX request. This logic is similar to other cases in the same method where `window.confirm` or `window.prompt` are used (e.g. [L783-L789](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/assets/js/admin/meta-boxes-product-variation.js#L783-L789) or [L792-L805](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/assets/js/admin/meta-boxes-product-variation.js#L792-L805)).

Alternatively, in analogy to those other cases, the `value` property of the `data` object could be set to `null` by the handler callback to signify that the `do_variation_action` method should end prematurely. In that case, the check should be:

```php
if ( null === data.value ) {
    return;
}
``` 
although this would prevent developers from using `null` as an admissible value for `data.value` while still being able to cancel the current operation. For this reason, the use of `data.cancel === true` might be more flexibile.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: plugins/woocommerce/assets/js/admin/meta-boxes-product-variation.js [hunk 1]
```diff
@@ -807,6 +807,10 @@ jQuery( function( $ ) {
 				default :
 					$( 'select.variation_actions' ).trigger( do_variation_action );
 					data = $( 'select.variation_actions' ).triggerHandler( do_variation_action + '_ajax_data', data );
+					
+					if ( true === data.cancel ) {
```
Comment (jeffstieler) [inline plugins/woocommerce/assets/js/admin/meta-boxes-product-variation.js]: Let's change this to a `null` check on `data` instead.

**Review by jeffstieler — CHANGES_REQUESTED** [review]
Thanks for the contribution!

After some discussion with the team, we think a strict `null` check on the `data` return from `triggerHandler()` would be best (rather than introducing a new `cancel` property and potentially breaking extensions that are already adding a `cancel` property that is meant to be sent in the AJAX request).

**Review by LuigiPulcini — COMMENTED** [review]
I changed my proposed code as per your request.

File: plugins/woocommerce/assets/js/admin/meta-boxes-product-variation.js (hunk 1 above)
Comment (LuigiPulcini) [reply plugins/woocommerce/assets/js/admin/meta-boxes-product-variation.js]: I changed this as per your request.

**Review by Konamiman — APPROVED** [review]

**Comment by Konamiman** [comment]
Merging since the requested change was implemented.

**Pull request merged**